### PR TITLE
limit attribute of IOContext is used for html generation

### PR DESCRIPTION
**Pull request by jw3126**
See #1098

### CODE REVIEW

File: src/abstractdataframe/io.jl [hunk 1]
```diff
@@ -178,8 +177,14 @@ end
         write(io, "<th>$column_name</th>")
     end
     write(io, "</tr>")
-    tty_rows, tty_cols = _displaysize(io)
-    mxrow = min(n,tty_rows)
+    haslimit = get(io, :limit, true)
+    n = size(df, 1)
+    if haslimit
+        tty_rows, tty_cols = displaysize(io)
```
Comment (nalimilan) [inline src/abstractdataframe/io.jl]: Use `_displaysize` for compatibility with 0.4.

File: test/io.jl [hunk 2]
```diff
@@ -516,4 +516,15 @@ module TestIO
     show(io, "text/html", df)
     str = takebuf_string(io)
     @test str == "<table class=\"data-frame\"><tr><th></th><th>Fish</th><th>Mass</th></tr><tr><th>1</th><td>Suzy</td><td>1.5</td></tr><tr><th>2</th><td>Amir</td><td>#NULL</td></tr></table>"
+
+    # test limit attribute of IOContext is used
+    df = DataFrame(a=collect(1:1000))
+    ioc = IOContext(IOBuffer(), displaysize=(10, 10), limit=false)
+    show(ioc, MIME{Symbol("text/html")}(), df)
```
Comment (nalimilan) [inline test/io.jl]: I think `MIME"text/html"` is equivalent and shorter.

File: test/io.jl [hunk 3]
```diff
@@ -516,4 +516,15 @@ module TestIO
...
+    @test ioc.io |> takebuf_string |> length > 10000
```
Comment (nalimilan) [inline test/io.jl]: Use standard syntax for nested function calls, as it's used elsewhere in the code. Also, maybe you could use `countlines` for a more precise check?

Comment (jw3126) [inline test/io.jl]: I just checked, there are currently no linebreaks in the html anyway.

**Comment by jw3126** [comment]
The default is the same as before (e.g. coupled to displaysize). Only if you set IOContext limit=false the output will be unlimited.

File: src/abstractdataframe/io.jl (hunk 1 above)
Comment (nalimilan) [inline src/abstractdataframe/io.jl]: Yeah, but adding an underscore doesn't hurt, and we could imagine backporting things. If we remove 0.4 support, we should remove `_displaysize` too, not just stop using it.

Comment (ararslan) [inline src/abstractdataframe/io.jl]: Right

**Pull request merged**